src/compas_timber/connections/plate_joint.py

```diff
@@ -212,6 +216,31 @@ def calculate_topology(self, allow_reordering=False):
         self.b_segment_index = topo_results.b_segment_index
         return topo_results
 
+    @classmethod
+    def promote_joint_candidate(cls, model, candidate, elements=None, **kwargs):
+        """Creates an instance of this joint from a generic joint.
+
+        Parameters
+        ----------
+        model : :class:`~compas_timber.model.TimberModel`
+            The model to which the elements and this joint belong.
+        candidate : :class:`~compas_timber.connections.Joint`
+            The generic joint to be converted.
+        elements : list(:class:`~compas_model.elements.Element`), optional
+            The elements to be connected by this joint. If not provided, the elements of the generic joint will be used.
+            This is used to explicitly define the element order.
+        **kwargs : dict
+            Additional keyword arguments that are passed to the joint's constructor.
+
+        Returns
+        -------
+        :class:`compas_timber.connections.Joint`
+            The instance of the created joint.
+
+        """
+        kwargs.update(candidate.__data__)  # pass topology and segment indices from candidate
```

Comment:
better explicit than implicit
```suggestion
        kwargs.update({topology=candidate.topology, a_segment_index=candidate.a_segment_index, b_segment_index=candidate.b_segment_index})
```